cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/QueryCoveringIndexesTest.java

```diff
@@ -115,6 +116,32 @@ public void returnsNullWhenQueryContainsInvalidValue() throws Exception {
     }
 
     @Test
+    public void returnsDocForQueryAgainstIndexWithSpecialChar() throws Exception {
+        DocumentRevision rev = new DocumentRevision("mike12");
+        Map<String, Object> bodyMap = new HashMap<String, Object>();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 12);
+        bodyMap.put("pet", "cat");
+        rev.setBody(DocumentBodyFactory.create(bodyMap));
+        ds.createDocumentFromRevision(rev);
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("name", "age"), "basic index"), is
+                ("basic index"));
```

Comment:
Should there be additional test cases for special character single quote, colon quote, brackets, and keyword?